src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,250 @@
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                foundedCar = (foundedCar == null)
+                        ? buildCarFromResultSet(resultSet)
+                        : addDriverToCar(buildDriverFromResultSet(resultSet), foundedCar);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get driver by id " + id, e);
+        }
+        return Optional.ofNullable(foundedCar);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        List<Car> cars = new ArrayList<>();
+        String getAllRequest = "SELECT c.id AS car_id, c.model, m.id AS manufacturer_id, "
+                + "m.name AS manufacturer_name, m.country,"
+                + "d.id AS driver_id, d.name AS driver_name, d.license_number "
+                + "FROM cars c "
+                + "JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "JOIN cars_drivers cd ON c.id = cd.cars_id "
+                + "JOIN drivers d ON d.id = cd.drivers_id "
+                + "WHERE c.is_deleted = FALSE AND d.is_deleted = FALSE;";
+        Car currentCar = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(getAllRequest)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                if (currentCar == null) {
+                    currentCar = buildCarFromResultSet(resultSet);
+                } else if (currentCar.getId().equals(resultSet.getLong("car_id"))) {
+                    addDriverToCar(buildDriverFromResultSet(resultSet), currentCar);
+                } else {
+                    cars.add(currentCar);
+                    currentCar = buildCarFromResultSet(resultSet);
+                }
+            }
+            cars.add(currentCar);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all cars from DB", e);
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateRequest = "UPDATE cars "
+                + "SET manufacturer_id = ?, model = ? "
+                + "WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                        = connection.prepareStatement(updateRequest)) {
+            statement.setLong(1, car.getManufacturer().getId());
+            statement.setString(2, car.getModel());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update "
+                    + car + " in cars DB.", e);
+        }
+        deletePreviousRelations(car.getId());
+        setCarToDriversRelation(car);
```

Comment:
```suggestion
        deleteAllDrivers(car.getId());
        insertAllDrivers(car);
```
I'd recommend renaming these methods this way